src/rust/ide/src/controller/visualisation.rs

```diff
@@ -0,0 +1,166 @@
+//! Visualisation controller.
+//!
+//! Ths Visualisation Controller is Responsible identifying all the available visualisation natively
+//! embedded in IDE and available within the project's `visualisation` folder. The
+//! `Visualisation Controller` lives as long as the `Project Controller`.
+
+use crate::prelude::*;
+
+use std::rc::Rc;
+use enso_protocol::language_server;
+
+
+
+// =============
+// === Error ===
+// =============
+
+/// Enumeration of errors used in `Visualisation Controller`.
+#[derive(Debug,Fail)]
+#[allow(missing_docs)]
+pub enum VisualiserError {
+    #[fail(display = "Visualiser \"{}\" not found.", identifier)]
+    NotFound {
+        identifier : VisualiserIdentifier
+    }
+}
+
+
+
+// ===============================
+// === VisualisationIdentifier ===
+// ===============================
+
+/// This enum is used to identify a visualiser both in the project folder or natively embedded in
+/// IDE.
+#[derive(Debug,Clone,PartialEq,Eq,Display)]
+#[allow(missing_docs)]
+pub enum VisualiserIdentifier {
+    Embedded(String),
```

Comment:
Don't we agreed with thar? Embedded vis are as a workaround - they are vis provided by us to be able in each project, because we won't support reading vis from "common place for all projects" until alpha at least.